src/my-bot.ts

```diff
@@ -1,38 +1,132 @@
-        return { Row: row, Column: column }
+    private getHitMap(shots): { [ pos: string] : boolean } {
+        var hitMap : { [ pos: string] : boolean } = {};
+        for(var i = 0; i < shots.length; ++i) {
+            var pos = new Position(shots[i].Position.Row, shots[i].Position.Column);
+            var posString = pos.getString();
+            if(shots[i].WasHit) hitMap[posString] = true;
+        }
+        return hitMap;
+    }
+
+    private getTriedMap(shots): { [ pos: string] : boolean } {
+        var triedMap : { [ pos: string] : boolean } = {};
+        for(var i = 0; i < shots.length; ++i) {
+            var pos = new Position(shots[i].Position.Row, shots[i].Position.Column);
+            var posString = pos.getString();
+            triedMap[posString] = true;
+        }
+        return triedMap;
     }
 
-    private getNextRow(row) {
-        var newRow = row.charCodeAt(0) + 1;
-        if(newRow > 'J'.charCodeAt(0)) {
-            return 'A';
+
+    //function which marks as tried (but not hits) the squares around two consecutive squares that we hit
+    private tryToMatchMoreAsTried(pos: Position, triedMap: { [ pos: string] : boolean }, hitMap: { [ pos: string] : boolean }) : { [ pos: string] : boolean } {
+        var upPos = pos.getUpPosition();
+        var upString = (upPos) ? (upPos.getString()) : "";
+        var downPos = pos.getDownPosition();
+        var downString = (downPos) ? (downPos.getString()) : "";
+        var leftPos = pos.getLeftPosition();
+        var leftString = (leftPos) ? (leftPos.getString()) : "";
+        var rightPos = pos.getRightPosition();
+        var rightString = ((rightPos) ? (rightPos.getString()) : "");
+        var thisString = pos.getString();
+        if(hitMap[thisString] && hitMap[upString]) {
+            triedMap = this.markLeftRight(pos, triedMap);
+            triedMap = this.markLeftRight(upPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(upPos, triedMap);
+        }
+        if(hitMap[thisString] && hitMap[downString]) {
+            triedMap = this.markLeftRight(pos, triedMap);
+            triedMap = this.markLeftRight(downPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(downPos, triedMap);
+        }
+        if(hitMap[thisString] && hitMap[leftString]) {
+            triedMap = this.markUpDown(pos, triedMap);
+            triedMap = this.markUpDown(leftPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(leftPos, triedMap);
+        }
+        if(hitMap[thisString] && hitMap[rightString]) {
+            triedMap = this.markUpDown(pos, triedMap);
+            triedMap = this.markUpDown(rightPos, triedMap);
+            triedMap = this.markDiagonal(pos, triedMap);
+            triedMap = this.markDiagonal(rightPos, triedMap);

```

Comment:
So, you're basically calling the up-down method twice, and the left-right method twice., just with two vars swapped.

This is a Code Smell - repetition is something we try to avoid as much as is possible. The reason is that, if we repeat things, it's very easy for someone to miss an instance, and get something wrong.

In this case, I think we could very easily set up a `tryUpDown(pos, triedMap, otherPos)` method, and a `tryLeftRight(pos, triedMap, otherPos)` one, cutting each of these ifs to a single line, and removing the duplication!